src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -977,7 +977,7 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
     const pkTextTracks = this._getPKTextTracks();
     const pkOffTrack = pkTextTracks.find(track => track.language === 'off');
     const getActiveVidTextTrackIndex = () => {
-      return this._nativeTextTracksMap.findIndex(textTrack => textTrack && this._getDisplayTextTrackModeString() === textTrack.mode);
+      return Object.values(this._nativeTextTracksMap).findIndex(textTrack => textTrack && this._getDisplayTextTrackModeString() === textTrack.mode);
```

Comment:
use the index/key from the map instead of the index in the array which is not relevant